File: source/clibCoreCs/Models/Entity/addonTemplateRules.cs
//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using System;
using System.Collections.Generic;
using System.Text;
using Contensive.BaseClasses;
using Contensive.Core.Controllers;
using Newtonsoft.Json;

namespace Contensive.Core.Models.Entity
{
	public class addonTemplateRuleModel : baseModel
	{
		//
		//====================================================================================================
		//-- const
		public const string contentName = "add-on template rules";
		public const string contentTableName = "ccAddonTemplateRules";
		private const string contentDataSource = "default";
		//
		//====================================================================================================
		// -- instance properties
		public int addonId {get; set;}
		public int templateId {get; set;}
		//
		//====================================================================================================
		public static addonTemplateRuleModel add(coreClass cpCore)
		{
			return add<addonTemplateRuleModel>(cpCore);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel add(coreClass cpCore, ref List<string> callersCacheNameList)
		{
			return add<addonTemplateRuleModel>(cpCore, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel create(coreClass cpCore, int recordId)
		{
			return create<addonTemplateRuleModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel create(coreClass cpCore, int recordId, ref List<string> callersCacheNameList)
		{
			return create<addonTemplateRuleModel>(cpCore, recordId, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel create(coreClass cpCore, string recordGuid)
		{
			return create<addonTemplateRuleModel>(cpCore, recordGuid);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel create(coreClass cpCore, string recordGuid, ref List<string> callersCacheNameList)
		{
			return create<addonTemplateRuleModel>(cpCore, recordGuid, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel createByName(coreClass cpCore, string recordName)
		{
			return createByName<addonTemplateRuleModel>(cpCore, recordName);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel createByName(coreClass cpCore, string recordName, ref List<string> callersCacheNameList)
		{
			return createByName<addonTemplateRuleModel>(cpCore, recordName, ref callersCacheNameList);
		}
		//
		//====================================================================================================
		public void save(coreClass cpCore)
		{
			base.save(cpCore);
		}
		//
		//====================================================================================================
		public static void delete(coreClass cpCore, int recordId)
		{
			delete<addonTemplateRuleModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static void delete(coreClass cpCore, string ccGuid)
		{
			delete<addonTemplateRuleModel>(cpCore, ccGuid);
		}
		//
		//====================================================================================================
		public static List<addonTemplateRuleModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList)
		{
			return createList<addonTemplateRuleModel>(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList);
		}
		//
		//====================================================================================================
		public static List<addonTemplateRuleModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy)
		{
			return createList<addonTemplateRuleModel>(cpCore, sqlCriteria, sqlOrderBy);
		}
		//
		//====================================================================================================
		public static List<addonTemplateRuleModel> createList(coreClass cpCore, string sqlCriteria)
		{
			return createList<addonTemplateRuleModel>(cpCore, sqlCriteria);
		}
		//
		//====================================================================================================
		public void invalidatePrimaryCache(coreClass cpCore, int recordId)
		{
			invalidateCacheSingleRecord<addonTemplateRuleModel>(cpCore, recordId);
		}
		//
		//====================================================================================================
		public static string getRecordName(coreClass cpcore, int recordId)
		{
			return baseModel.getRecordName<addonTemplateRuleModel>(cpcore, recordId);
		}
		//
		//====================================================================================================
		public static string getRecordName(coreClass cpcore, string ccGuid)
		{
			return baseModel.getRecordName<addonTemplateRuleModel>(cpcore, ccGuid);
		}
		//
		//====================================================================================================
		public static int getRecordId(coreClass cpcore, string ccGuid)
		{
			return baseModel.getRecordId<addonTemplateRuleModel>(cpcore, ccGuid);
		}
		//
		//====================================================================================================
		public static addonTemplateRuleModel createDefault(coreClass cpcore)
		{
			return createDefault<addonTemplateRuleModel>(cpcore);
		}
	}
}

File: source/Processor/Models/Db/0/addonTemplateRules.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class addonTemplateRuleModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "add-on template rules"
        Public Const contentTableName As String = "ccAddonTemplateRules"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        Public Property addonId As Integer
        Public Property templateId As Integer
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As addonTemplateRuleModel
            Return add(Of addonTemplateRuleModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As addonTemplateRuleModel
            Return add(Of addonTemplateRuleModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As addonTemplateRuleModel
            Return create(Of addonTemplateRuleModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As addonTemplateRuleModel
            Return create(Of addonTemplateRuleModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String) As addonTemplateRuleModel
            Return create(Of addonTemplateRuleModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As addonTemplateRuleModel
            Return create(Of addonTemplateRuleModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As addonTemplateRuleModel
            Return createByName(Of addonTemplateRuleModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As addonTemplateRuleModel
            Return createByName(Of addonTemplateRuleModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of addonTemplateRuleModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of addonTemplateRuleModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String, callersCacheNameList As List(Of String)) As List(Of addonTemplateRuleModel)
            Return createList(Of addonTemplateRuleModel)(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String) As List(Of addonTemplateRuleModel)
            Return createList(Of addonTemplateRuleModel)(cpCore, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of addonTemplateRuleModel)
            Return createList(Of addonTemplateRuleModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidateCacheSingleRecord(Of addonTemplateRuleModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of addonTemplateRuleModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of addonTemplateRuleModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of addonTemplateRuleModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As addonTemplateRuleModel
            Return createDefault(Of addonTemplateRuleModel)(cpcore)
        End Function
    End Class
End Namespace
